lib/os/heap.c

```diff
@@ -8,6 +8,19 @@
 #include <string.h>
 #include "heap.h"
 
+static inline void do_callback(struct z_heap *h,
+			       enum sys_heap_cb_ops op,
+			       void *mem, size_t bytes)
+{
+#ifdef CONFIG_SYS_HEAP_ALLOC_FREE_CALLBACKS
+	struct sys_heap *sh = CONTAINER_OF(h, struct sys_heap, heap);
+
+	if (sh->alloc_free_cb != NULL) {
+		sh->alloc_free_cb(op, mem, bytes);
+	}
+#endif
+}
```

Comment:
I don't see this being used anywhere. Remove?

Reply:
Removed.